modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -3826,10 +3952,14 @@ private void assignPartitionsStates() {
     }
 
     /**
-     * Removes gaps in the local update counters. Gaps in update counters are possible on backup node when primary
-     * failed to send update counter deltas to backup.
+     * Removes gaps in the local update counters. Affects specified node's primary partitions in case node specified and
+     * all local partitins otherwise.
+     *
+     * Gaps in update counters are possible on backup node when primary failed to send update counter deltas to backup.
+     *
+     * @param nodeId Failed node id, {@code null} in case of merged failures.
      */
-    private void finalizePartitionCounters() {
+    private void finalizePartitionCounters(UUID nodeId) {
```

Comment:
@Nullable is missing

Reply:
What's the reason for using this annotation? 
Everything in nullable until checked to be not null.

Reply:
To emphasize special meaning of null argument. We have lots of use of @Nullable annotation in the codebase.

Reply:
Not sure I understand who checks them... and how?
What contract we have here, what difference with absent annotation?
Seems, that's a holywar issue :)
Will you mind if I keep this as is?

Reply:
I wouldn't mind )